### ISSUE TITLE
val-acc always 0.000

### ISSUE THREAD

**Issue by sh1man** [kwon-evan/LPRNet#7]
I started the training with the config.yaml configuration
Have you tried starting a workout from scratch?
val-acc always 0

`Epoch 88:  91%|█████████ | 396/437 [00:38<00:03, 10.27it/s, loss=inf, v_num=3g20, train-loss=inf.0, train-acc=0.000, val-loss=inf.0, val-acc=0.000]
Epoch 88:  91%|█████████ | 397/437 [00:38<00:03, 10.28it/s, loss=inf, v_num=3g20, train-loss=inf.0, train-acc=0.000, val-loss=inf.0, val-acc=0.000]
Epoch 88:  91%|█████████ | 398/437 [00:38<00:03, 10.29it/s, loss=inf, v_num=3g20, train-loss=inf.0, train-acc=0.000, val-loss=inf.0, val-acc=0.000]
Epoch 88:  91%|█████████▏| 399/437 [00:38<00:03, 10.30it/s, loss=inf, v_num=3g20, train-loss=inf.0, train-acc=0.000, val-loss=inf.0, val-acc=0.000]`

**Comment by kwon-evan** [comment]
It seems that loss is infinite in both training and validation.
There are a number of things you can do to fix this, but how about reducing the learning rate?

**Comment by kwon-evan** [comment]
I tested it while updating the code(#8), and there was no problem with training. Can you check your training settings?

**Comment by sh1man** [comment]
Could there be a problem with the Cyrillic characters in the image title?

**Comment by kwon-evan** [comment]
Oh, Yes. All characters must be listed in the config.

**Comment by sh1man** [comment]
Strange, but training worked here [https://github.com/kwon-evan/deep-text-recognition-benchmark](https://github.com/kwon-evan/deep-text-recognition-benchmark)

**Comment by sh1man** [comment]
> Oh, Yes. All characters must be listed in the config.

yes I indicated everything

**Comment by kwon-evan** [comment]
Did you try reducing learning rate? or you can use this [callback](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.LearningRateFinder.html#lightning.pytorch.callbacks.LearningRateFinder) to find lr.

**Comment by sh1man** [comment]
For some reason the problem is in the name of the image labels, if you change their name to the English layout, then everything is fine.

> Did you try reducing learning rate? or you can use this [callback](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.LearningRateFinder.html#lightning.pytorch.callbacks.LearningRateFinder) to find lr.

**Comment by sh1man** [comment]
If you name files like this, it won't work
```
chars: [
    # NUMBERS
    '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
    # ALPHABETS
    'А','В','Е','К','М','Н','О','Р','С','Т','У','Х'
]
```

**Comment by kwon-evan** [comment]
> If you name files like this, it won't work
> 
> ```
> chars: [
>     # NUMBERS
>     '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
>     # ALPHABETS
>     'А','В','Е','К','М','Н','О','Р','С','Т','У','Х'
> ]
> ```

Is this character you want to train? If you get any errors, can you show me error msg? and some of your data labels?

**Comment by sh1man** [comment]
Here's a small dataset. There are no errors, always train-loss=inf.0, train-acc=0.000, val-loss=inf.0, val-acc=0.000.
dataset
https://drive.google.com/file/d/1QZwLUr-74KTbaijCErtYsJXKXSviQOKK/view?usp=sharing
or
https://github.com/sh1man/dataset_light

**Comment by sh1man** [comment]
t_length: 18

The error was in this parameter, I changed it from 18 to 9. What does this parameter affect ?

**Comment by sh1man** [comment]
Should it be like that?
I have 55 thousand training data

Epoch 0: 100%|██████████| 790/790 [00:43<00:00, 18.34it/s, v_num=6iga, train-loss=0.00245, train-acc=0.714]
Epoch 1: 100%|██████████| 790/790 [00:46<00:00, 17.05it/s, v_num=6iga, train-loss=0.139, train-acc=0.619, val-loss=0.206, val-acc=0.687]Metric val-acc improved by 0.041 >= min_delta = 0.0. New best score: 0.687
Epoch 2: 100%|██████████| 790/790 [00:46<00:00, 16.95it/s, v_num=6iga, train-loss=0.0634, train-acc=0.857, val-loss=0.168, val-acc=0.698]Metric val-acc improved by 0.011 >= min_delta = 0.0. New best score: 0.698
Epoch 3: 100%|██████████| 790/790 [00:47<00:00, 16.55it/s, v_num=6iga, train-loss=0.0489, train-acc=0.714, val-loss=0.172, val-acc=0.691]Epoch 3, global step 3160: 'val-acc' reached 0.69085 (best 0.69843), saving model to 'train_weights\\epoch=03-val-acc=0.691.ckpt' as top 5

**Comment by kwon-evan** [comment]
Yeah, It seems to train properly. I think t_length is used in [here](https://github.com/kwon-evan/LPRNet/blob/45bf3be36a74221dc6bf954570a6abfa32df422a/lprnet/lprnet.py#L14) for F.ctc_loss. I'm not sure about the loss calculation, so please check [here](https://pytorch.org/docs/stable/generated/torch.nn.functional.ctc_loss.html#torch.nn.functional.ctc_loss).

**Comment by sh1man** [comment]
how to use attn instead of ctc?

unfortunately, max train-acc=0.771 by the end of epoch 100, training was not successful.

**Comment by sh1man999** [comment]
Is it possible to improve the model's accuracy results? What parameters should be changed?
I ended up with train-acc=0.821

**Comment by kwon-evan** [comment]
Your data may be too difficult for LPRNet.
I'm afraid I haven't tried to improve the accuracy of the model, so I can't help you there.

**Comment by sh1man** [comment]
I sent you the dataset, there’s nothing complicated there. 
https://github.com/sh1man/dataset_light

**Comment by kwon-evan** [comment]
Hi @sh1man, I have just approved PR #9 #10.
Now augmentation during training is available.
I think it would be helpful to improve accuracy.

**Comment by sh1man999** [comment]
Thanks, I'll try it today and publish the results

**Comment by sh1man999** [comment]
It would be interesting to replace СTС with ATTN. ATTN in this project was accurate. https://github.com/kwon-evan/deep-text-recognition-benchmark

**Comment by sh1man** [comment]
What is the minimum dataset required?
I'm trying to train on 50,000 images. This doesn't work. 
Approximately how many epochs are needed?
Epoch 12: 100%|██████████| 97/97 [01:32<00:00,  1.05it/s, v_num=4suy, train-loss=0.406, train-acc=0.504, val-loss=1.780, val-acc=0.000224]Epoch 12, global step 1261: 'val-acc' reached 0.00022 (best 0.00494), saving model to 'train_weights\\epoch=12-val-acc=0.000.ckpt' as top 5

**Comment by sh1man** [comment]
I achieved without data_augment.
I would be interested to know about square plate recognition.

Epoch 40: 100%|██████████| 97/97 [01:11<00:00,  1.36it/s, v_num=e2qp, train-loss=0.00107, train-acc=1.000, val-loss=0.000353, val-acc=1.000]Epoch 40, global step 3977: 'val-acc' reached 0.99955 (best 0.99955), saving model to 'train_weights\\epoch=40-val-acc=1.000.ckpt' as top 5

**Comment by sh1man** [comment]
the number of characters affects the accuracy why?
I don't use certain characters in my dataset.
I used this chars and it was always val-acc=0.000
chars: [
    # NUMBERS
    '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
    # ALPHABETS
    'A','B','C','E','K','M','H','O','P','T','Y','X',
]

**Comment by sh1man** [comment]
Is there information?

**Comment by kwon-evan** [comment]
Sorry for late.
 
> It would be interesting to replace СTС with ATTN. ATTN in this project was accurate. https://github.com/kwon-evan/deep-text-recognition-benchmark

I'm working on other things and don't plan to maintain this model anymore, so if you'd like to contribute by implementing it yourself, that would be great.

> the number of characters affects the accuracy why?
> I don't use certain characters in my dataset.
> I used this chars and it was always val-acc=0.000 

As far as I know, the number of characters doesn't affect accuracy, so I'm afraid I can't help you with this one.

And as mentioned above, this repository is hard to maintain because of other things. I'll close this issue for now, but feel free to reopen it if needed.

**Comment by sh1man999** [comment]
Can you suggest which models are more relevant and work better now?

**Comment by kwon-evan** [comment]
As an alternative to LPRNet, I recommend [ViTSTR](https://github.com/roatienza/deep-text-recognition-benchmark). It's almost as fast as LRPNet and more accurate.
